ci: `tree-editor` template fails CI

**Description**

The `tree-editor` template has been failing for some time now in CI resulting in a failed build:
- https://github.com/eclipse-theia/generator-theia-extension/runs/5787095655?check_suite_focus=true#step:5:2732

```
Error: tree-editor: src/browser/example-file/example-file-command.ts(49,78): error TS2345: Argument of type 'URI' is not assignable to parameter of type 'FileStat'.
tree-editor:   Type 'URI' is missing the following properties from type 'FileStat': isFile, isDirectory, isSymbolicLink, resource, name
```

From what I can tell it has been failing since 16 days ago: https://github.com/eclipse-theia/generator-theia-extension/runs/5578246651?check_suite_focus=true#step:5:2756.

cc @JonasHelming

--- a/templates/tree-editor/example-file/example-file-command.ts
+++ b/templates/tree-editor/example-file/example-file-command.ts
@@ -45,8 +45,8 @@ export class NewTreeExampleFileCommandHandler implements SingleUriCommandHandler
             return;
         }
 
-        const dirUri = dir.resource;
-        const preliminaryFileUri = FileSystemUtils.generateUniqueResourceURI(dirUri, dir, 'tree-example', '.tree');
+        const targetUri = dir.resource.resolve('tree-example.tree');
+        const preliminaryFileUri = FileSystemUtils.generateUniqueResourceURI(dir, targetUri, false);
         const dialog = new SingleTextInputDialog({
             title: 'New Example File',
             initialValue: preliminaryFileUri.path.base
@@ -54,7 +54,7 @@ export class NewTreeExampleFileCommandHandler implements SingleUriCommandHandler
 
         const fileName = await dialog.open();
         if (fileName) {
-            const fileUri = dirUri.resolve(fileName);
+            const fileUri = dir.resource.resolve(fileName);
             const contentBuffer = BinaryBuffer.fromString(JSON.stringify(defaultData, null, 2));
             this.fileService.createFile(fileUri, contentBuffer)
                 .then(_ => this.openerService.getOpener(fileUri))

